crates/uv-cli/src/lib.rs

```diff
@@ -2606,30 +2606,83 @@ pub struct ToolNamespace {
 
 #[derive(Subcommand)]
 pub enum ToolCommand {
-    /// Run a tool.
+    /// Run a command provided by a Python package.
+    ///
+    /// By default, the package to install is assumed to match the command name.
+    ///
+    /// The name of the command can include an exact version in the format
+    /// `<package>@<version>`, e.g., `uv run ruff@0.3.0`. If more complex
+    /// version specification is desired or if the command is provided by a
+    /// different package, use `--from`.
+    ///
+    /// If the tool was previously installed, i.e., via `uv tool install`, the
+    /// installed version will be used unless a version is requested or the
+    /// `--isolated` flag is used.
+    ///
+    /// A `uvx` alias is provided for `uv tool run` for quick invocations.
+    ///
+    /// If no command is provided, the installed tools are displayed.
+    ///
+    /// Packages are installed into an ephemeral virtual environment in the uv
+    /// cache directory.
+    ///
+    /// Be careful when invoking tools, package installation can execute
```

Comment:
"when invoking tools, as package installation..."

(Is this piece necessary?)

Reply:
I'm a little hesitant on it, but I want to warn about dependency confusion attacks e.g. `uvx foobar` when `foobar` is actually provided by an official `foobar-cli` is a real risk.